test/react-native/features/support/env.rb

```diff
@@ -39,6 +39,10 @@ def device_type
   skip_this_scenario("Skipping scenario") if $driver.capabilities["os"] == 'android'
 end
 
+Before('@navigation') do |scenario|
+  skip_this_scenario("Skipping scenario") unless ENV['RUN_NAVIGATION_SCENARIOS']
```

Comment:
I'd rather have a `SKIP_NAVIGATION_SCENARIOS` environment variable so that we don't end up running nothing if we forget to set it.